### PR TITLE
Add: mergeAll

### PR DESCRIPTION
**Pull request by jordalgo**

### CODE REVIEW

**Comment by jordalgo** [comment]
This is more of an RFC. Just wanted to see if this was a good path before I write the unit tests and documentation for it. 

It could also be made curried instead by requiring the first param to be the number of expected streams to be merged.

File: lib/index.js [hunk 1]
```diff
@@ -45,6 +45,26 @@ var merge = curryN(2, function(s1, s2) {
...
+    [].slice.call(arguments, streams.length).some(function(s1) {
+      if (s1.hasVal) {
+        self(s1.val);
+        return true;
+      }
+    });
+  }, streams));
+  endsOn(combine(function() {
+    return true;
+  }, streams.map(function(sm) { return sm.end; })), s);
```
Comment (paldepind) [inline lib/index.js]: Maybe the merged stream should only end once all the streams have ended?

**Comment by paldepind** [comment]
This would be a fine function to have :+1: But I think it should be included as a module an not as part of the core Flyd file.

File: lib/index.js (hunk 1 above)
Comment (jordalgo) [inline lib/index.js]: Isn't that what is happening here? I've tested locally and I believe this is the same logic used for `merge`.

**Comment by jordalgo** [comment]
No problem. I'll push up a separate PR where it's a module instead. Nice work on the lib btw!

File: lib/index.js (hunk 1 above)
Comment (paldepind) [inline lib/index.js]: Yes. Indeed. You are correct.

**Pull request closed without merging**